cannon/mipsevm/memory/hasher.go

```diff
@@ -0,0 +1,74 @@
+package memory
+
+import (
+	"sync"
+
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// Byte32Pool is a sync.Pool for [32]byte slices
+var Byte32Pool = sync.Pool{
+	New: func() interface{} {
+		var b [32]byte
+		return &b // Return a pointer to avoid extra allocations
+	},
+}
+
+// GetByte32 retrieves a *[32]byte from the pool
+func GetByte32() *[32]byte {
+	return Byte32Pool.Get().(*[32]byte)
+}
+
+// PutByte32 returns a *[32]byte to the pool
+func PutByte32(b *[32]byte) {
```

Comment:
(naming nit) Might be clearer with something like:
```suggestion
func ReleaseByte32(b *[32]byte) {
```